client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -404,11 +404,7 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 
 	// When the intent is build, we can potentially show the generated design picker.
 	// Don't render until we've fetched the generated designs from the backend.
-	if (
-		enabledGeneratedDesigns &&
-		( isLoadingGeneratedDesigns || ! siteVerticalId ) &&
-		! isAnchorSite
-	) {
+	if ( enabledGeneratedDesigns && isLoadingGeneratedDesigns ) {
```

Comment:
hmmm...in this way, if you refresh the page, you will see the static design picker and then generated design picker when the site vertical id is ready.

Reply:
I thought so too, but I think `isLoadingGeneratedDesigns` will only be true when  `siteVerticalId` is available. Since the condition for `useStarterDesignsGeneratedQuery` to be enabled is  `enabledGeneratedDesigns && !! siteVerticalId` 😅 

Taking a deeper look, I don't think we need `enabledGeneratedDesigns` here either since it's also built in the condition above. We should only need  `isLoadingGeneratedDesigns` in this line. That would make things simpler, I think!

Reply:
> but I think isLoadingGeneratedDesigns will only be true when siteVerticalId is available.

Yes, so the step shows Static Design Picker first when `siteVerticalId` is not available and renders null when `siteVerticalId` is available. In the end, it shows Generated Design Picker after the generated design is loaded.


https://user-images.githubusercontent.com/13596067/171318834-b119b686-2fc5-4348-ba48-ea214c6834fa.mov

(You can see the nav buttons shown at the beginning (but the static designs is still loading so that the screen is still white) and disappear later

Reply:
Nice find @taipeicoder.

@arthur791004 I added console logs and see that you are correct that it will attempt to render the static design picker briefly at first as `isLoadingGeneratedDesigns` is false but `showGeneratedDesigns` is also false.

I'm looking into using the `isFetched` property of the react Query object rather than `isLoading`.

Reply:
I think we have to check whether the site is ready or not

```js
if ( ! site || isLoadingGeneratedDesigns ) {
  return null;
}
```

Reply:
hmm so as you mentioned @arthur791004 the problem is that the site vertical id takes several renders to become ready when loading the page for the first time. I think to solve that we would also need a loading state for site vertical id.

Fortunately, site vertical id only needs to load when the page is being refreshed. When you navigate through the flow normally, without refreshing, it should already be loaded by the time you land on the page. So I don't think it's really an issue.

Reply:
What @arthur791004 proposes makes sense and it's simple. We want to essentially block rendering in two scenarios:

1. When determining whether we should render static or generated design picker (we need `site` for this).
2. If generated design picker, then we need to determine whether recipes are returned or not (we need `isLoadingGeneratedDesigns` to be false)

Reply:
> I think to solve that we would also need a loading state for site vertical id.

I think we just need to use `site` to check the loading state for site vertical id as site vertical id is from site options. When the `site` is ready, the site vertical id is also available.

Reply:
oops sorry I commented without refreshing to see your latest comments 😕
Nice, checking for `! site` does the trick!